File: frontend/src/services/authService.ts
import { apiClient, unwrap } from '../api/client'
import type { AuthResponse, LoginRequest, RegisterRequest } from '../api/types'

export const authService = {
  async login(request: LoginRequest): Promise<AuthResponse> {
    return unwrap(await apiClient.post('/auth/login', request))
  },

  async register(request: RegisterRequest): Promise<AuthResponse> {
    return unwrap(await apiClient.post('/auth/register', request))
  },

  async refresh(refreshToken: string): Promise<AuthResponse> {
    return unwrap(await apiClient.post('/auth/refresh', { refreshToken }))
  },

  async logout(refreshToken: string): Promise<void> {
    await apiClient.post('/auth/logout', { refreshToken })
  },
}

File: src/main/java/com/sarjeev/booktheshow/services/AuthService.java
package com.sarjeev.booktheshow.services;

import com.sarjeev.booktheshow.requests.LoginRequest;
import com.sarjeev.booktheshow.requests.LogoutRequest;
import com.sarjeev.booktheshow.requests.RefreshTokenRequest;
import com.sarjeev.booktheshow.requests.RegisterRequest;
import com.sarjeev.booktheshow.responses.AuthResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refresh(RefreshTokenRequest request);

    void logout(LogoutRequest request);
}
